### PR TITLE
RM14 - Map View

### PR DESCRIPTION
**Pull request by QAldo**
- Create Map View Fragment
- Retrieve current location
- Initialize sample locations for testing

### CODE REVIEW

File: app/src/main/java/csis3175/w23/g11/rooftown/roommates/ui/view/MapViewFragment.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+        if (ActivityCompat.checkSelfPermission(requireContext(), Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED){
+            mMap.setMyLocationEnabled(true);
+            mMap.getUiSettings().setMyLocationButtonEnabled(true);
+            fusedLocationProviderClient.getLastLocation().addOnSuccessListener(requireActivity(), new OnSuccessListener<Location>() {
+                @Override
+                public void onSuccess(Location location) {
+                    if (location != null) {
+                        LatLng latLng = new LatLng(location.getLatitude(), location.getLongitude());
+                        mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(firstLocation, 15));
```
Comment (vinztsang) [inline app/src/main/java/csis3175/w23/g11/rooftown/roommates/ui/view/MapViewFragment.java:85]: When I was testing, I noticed several things, but just for discussing, not a must to update:
1. When opening the map view for the first time, the app will not have a popup to ask for authorization, and I need to go to the setting to set it by myself. Is it better to have a popup to ask?
2. Is it better to have a default start location if the app cannot retrieve the user's location?
3. I expect that user location should only use once when the map starts. But for now, it keeps moving the map to the user's current place, which seems inconvenient when browsing the map.

**Review by vinztsang — COMMENTED** [review]
@QAldo, maybe you can keep working on this branch and not merge to the main branch first. I will try branching from your branch to work on my part

File: app/src/main/java/csis3175/w23/g11/rooftown/ListViewFragment.java [hunk 2]
```diff
@@ -4,15 +4,39 @@
...
+            new LatLng(49.2263531659175, -122.99951733755651)   // Place 3
+    );
+
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
+        View rootView = inflater.inflate(R.layout.fragment_list_view, container, false);
+        ListView listViewPost = rootView.findViewById(R.id.listViewPost);
+        ArrayAdapter<LatLng> siteArrAdapter = new ArrayAdapter<>(requireContext(), android.R.layout.simple_list_item_1,locationList);
```
[the remark this answers is not in the document]
Comment (QAldo) [reply app/src/main/java/csis3175/w23/g11/rooftown/ListViewFragment.java:32]: Will work on this after map view

File: app/src/main/java/csis3175/w23/g11/rooftown/roommates/ui/view/MapViewFragment.java (hunk 1 above)
Comment (QAldo) [reply app/src/main/java/csis3175/w23/g11/rooftown/roommates/ui/view/MapViewFragment.java:90]: Fixed

**Comment by QAldo** [comment]
Screenshot_20230311_005955
Simply hard-code the info window box wording at the moment, will create object accordingly.

**Comment by QAldo** [comment]
Screenshot_20230311_010136
Simply hard-code the info window box wording at the moment, will create post object accordingly.

**Review by vinztsang — APPROVED** [review]
Looks good. Let's merge it first

**Pull request merged**